src/ompl/base/samplers/src/InformedStateSampler.cpp

```diff
@@ -0,0 +1,117 @@
+/*********************************************************************
+* Software License Agreement (BSD License)
+*
+*  Copyright (c) 2014, University of Toronto
+*  All rights reserved.
+*
+*  Redistribution and use in source and binary forms, with or without
+*  modification, are permitted provided that the following conditions
+*  are met:
+*
+*   * Redistributions of source code must retain the above copyright
+*     notice, this list of conditions and the following disclaimer.
+*   * Redistributions in binary form must reproduce the above
+*     copyright notice, this list of conditions and the following
+*     disclaimer in the documentation and/or other materials provided
+*     with the distribution.
+*   * Neither the name of the University of Toronto nor the names of its
+*     contributors may be used to endorse or promote products derived
+*     from this software without specific prior written permission.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+/* Authors: Jonathan Gammell */
+
+#include "ompl/base/samplers/InformedStateSampler.h"
+#include "ompl/util/Exception.h"
+#include "ompl/base/OptimizationObjective.h"
+// The goal definitions
+#include "ompl/base/Goal.h"
+
+namespace ompl
+{
+    namespace base
+    {
+        // The base InformedStateSampler class:
+        InformedStateSampler::InformedStateSampler(const StateSpace* space, const ProblemDefinitionPtr probDefn, const Cost* bestCost)
+          : StateSampler(space),
+            probDefn_(probDefn),
+            bestCostPtr_(bestCost)
+        {
+            // Sanity check the problem.
+            // Check that we were given a valid pointer
+            if (bestCostPtr_ == false)
+            {
+                throw Exception ("InformedStateSampler: The cost pointer must be valid at construction.");
+            }
+            // No else
+
+            // Check that there is an optimization objective
+            if (probDefn_->hasOptimizationObjective() == false)
```

Comment:
I don't think so?
The InformedSampler depends on the optimization objective. You would need a different sampler if you were optimizing path length (`DirectPathLengthInfSampler`) vs. path clearance (which could use the general `RejectionInfSampler`).